### PR TITLE
fix: typo in refresh_equipment

### PR DESCRIPTION
**Pull request by xavierd**
Reading the code, I noticed that an undefined method was being called.

### CODE REVIEW

File: src/pyeconet/api.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ async def refresh_equipment(self) -> None:
                 _equip_obj: Equipment = None
                 equipment = self._equipment.get(_equip.get("device_name", ""), None)
```
Comment (CodingSquirrel) [inline src/pyeconet/api.py:166]:
```suggestion
                serial = _equip.get("serial_number")
                equipment = self._equipment.get(serial)
```

`device_name` isn't the key for equipment, `serial_number` is. That explains why the below typo never throws for trying to call a non-existent function.

**Review by CodingSquirrel — CHANGES_REQUESTED** [review]
@w1ll1am23 Can we get some eyes on this PR? Home assistant integration calls this function to refresh values hourly, but it currently does nothing. @xavierd 's and my changes should fix that.

**Comment by w1ll1am23** [comment]
Nice! Good catch. Assuming y'all have tested this and its all working? I don't have the capacity to test it now, but I can get this merged and get a fix version out there.

**Comment by CodingSquirrel** [comment]
I didn't do a full test, just checked some of the lines in a terminal to check the keys matched serial, and double checked reading over the code. If you'd prefer better testing before merging I can do some more thorough testing.

**Comment by w1ll1am23** [comment]
Ehh we can merge it, doesn't break anything until someone bumps the Home Assistant version.

Is the change on the HA side something one of you will be opening? If so, you can just test it out when making that change.

**Comment by CodingSquirrel** [comment]
I think my suggested change got left out of the merged change. I can submit as a separate PR.

**Comment by w1ll1am23** [comment]
Ugh sorry yeah I forgot you left that comment my bad shouldn't have been looking at that on mobile.

**Pull request merged**